docker/entrypoint.py

```diff
@@ -6,28 +6,83 @@
 import base64
 import subprocess
 import shlex
+import shutil
+from pathlib import Path
+from zipfile import ZipFile
+from typing import List
 
+SLIDES: str = "\n" + ("=" * 40) + "\n"
 
-def extract_war_files(webapps_dir, webapps_ref, path_prefix):
-    """
-    Prepare webapps on first-time startup:
-    - Extract WAR files from webapps-ref to webapps.
-    - Modify log4j to send logs to stdout.
-    - Modify context path (pathPrefix) if given.
-
-    :param webapps_dir: Path to the webapps directory.
-    :param webapps_ref: Path to the webapps reference directory.
-    :param path_prefix: Prefix for the context path.
+
+def extract(source: Path, destination: Path, prefix: str = "/", kind: str = "war"):
+    """Extract a file to a destination directory.
```

Comment:
`extract` should be called only once during the initial startup. it looks like it's getting called on every startup.